### PR TITLE
Format to Formatter rename

### LINKED ISSUE #274: [naming][bikeshedding] `*Format` vs `*Formatter`
Why do we have `DateTimeFormat` and not `DateTimeFormatter`? This API is a formatter, not a format, and it produces `FormattedDateTime`.

Is it just copypasta from ECMA402? I see ICU4C having `DateTimeFormat` and `NumberFormat` and `NumberFormatter`...

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes #274

### CODE REVIEW

**Review by sffc — APPROVED** [review]
I'm happy to approve these renames component-by-component

File: provider/datagen/src/transform/cldr/time_zones/mod.rs [hunk 1]
```diff
@@ -110,7 +110,7 @@ macro_rules! impl_resource_provider {
 }
 
 impl_resource_provider!(
-    TimeZoneFormatsV1Marker,
+    TimeZoneFormattersV1Marker,
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/time_zones/mod.rs:113]: Nit: This is one that actually is supposed to be `Format`

File: ffi/diplomat/wasm/test/fixed-decimal-format.mjs [hunk 2]
```diff
@@ -4,11 +4,11 @@
 
 import test from 'ava';
 
-import { ICU4XFixedDecimal, ICU4XLocale, ICU4XDataProvider, ICU4XFixedDecimalFormat, ICU4XFixedDecimalFormatOptions } from "../lib/api.mjs"
+import { ICU4XFixedDecimal, ICU4XLocale, ICU4XDataProvider, ICU4XFixedDecimalFormatter, ICU4XFixedDecimalFormatterOptions } from "../lib/api.mjs"
```
Comment (robertbastian) [inline ffi/diplomat/wasm/test/fixed-decimal-format.mjs:7]: Rename this file?

File: ffi/diplomat/wasm/bench/fixed-decimal-format.mjs [hunk 3]
```diff
@@ -1,21 +1,21 @@
 import Benchmark from 'benchmark';
```
Comment (robertbastian) [inline ffi/diplomat/wasm/bench/fixed-decimal-format.mjs:1]: Rename this file?

File: provider/datagen/src/transform/cldr/time_zones/convert.rs [hunk 4]
```diff
@@ -43,7 +43,7 @@ fn parse_hour_format(hour_format: &str) -> (Cow<'static, str>, Cow<'static, str>
     (Cow::Owned(positive), Cow::Owned(negative))
 }
 
-impl From<&CldrTimeZonesData> for TimeZoneFormatsV1<'static> {
+impl From<&CldrTimeZonesData> for TimeZoneFormattersV1<'static> {
```
Comment (robertbastian) [inline provider/datagen/src/transform/cldr/time_zones/convert.rs:46]: You forgot this. There shouldn't be a diff in `provider/`

File: docs/posts/20210427_ICU4X_02_Release.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ The revision improves completeness of the components introduced in [ICU4X 0.1] a
 
 ## Components
 
-In 0.2, ICU4X introduces [`FixedDecimalFormat`] which serves as a minimal and lightweight number format API.
+In 0.2, ICU4X introduces [`FixedDecimalFormatter`] which serves as a minimal and lightweight number format API.
```
Comment (sffc) [inline docs/posts/20210427_ICU4X_02_Release.md:9]: Nit: I'd rather not change the historical names in the blog post archive

**Review by sffc — DISMISSED** [review]

**Review by sffc — APPROVED** [review]
Please fix the merge conflict

I will try and help you hit the merge button when it is clickable

**Review by sffc — COMMENTED** [review]
Can you also update the new APIs that @dminor added in newly-merged #2164?

**Review by sffc — APPROVED** [review]
let's do it!

**Pull request merged**